src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -151,6 +167,9 @@ private void checkOut(RoutingContext routingContext) {
       .thenComposeAsync(r -> r.after(relatedRecords -> checkOut(relatedRecords,
         routingContext.getBodyAsJson(), clients)))
       .thenApply(r -> r.map(this::checkOutItem))
+      .thenCompose(r -> r.combineAfter(settingsRepository::lookUpCheckOutLockSettings, LoanAndRelatedRecords::withCheckoutLockConfiguration))
+      .thenCompose(r -> r.after(records -> this.acquireLock(records, checkOutLockRepository, checkOutLockId)))
+      .thenCompose(r -> r.after(records -> this.validateItemLimitBasedOnLockFeatureFlag(records, validators, errorHandler)))
```

Comment:
Can these lines be extracted to a method? This way, we'll only have a single line responsible for a lock and avoid making "check out future chain" even longer than it already is.